statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DeleteHandler.java

```diff
@@ -33,9 +33,7 @@
 
             if (!currentContext.isDeletionStarted()) {
                 if (!doesStateMachineExist(model, proxy, sfnClient)) {
-                    return ProgressEvent.<ResourceModel, CallbackContext>builder()
-                            .status(OperationStatus.SUCCESS)
-                            .build();
+                    throw getStateMachineDoesNotExistException();
```

Comment:
Is this intentional? This is a behaviour change unrelated to dependency versions.

Reply:
This change was included as part of https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-stepfunctions/pull/21. I have merged the changes from `main` into the branch, and these are no longer included as part of the file diff.